### ISSUE TITLE
Cannot have both didTapMarker: and didTapCluster: / didTapClusterItem:

### ISSUE THREAD

**Issue by hansemannn** [googlemaps/google-maps-ios-utils#73]
Hey there,

here is my problem: I want to handle both the clicks on the cluster, but also on the marker itself (primary because I cannot find a way to get the `GMSMarker` in the `didTapClusterItem:` delegate.

So when I have both the `didTapMarker:` and one of `didTapCluster:` or `didTapClusterItem:` implemented, `didTapMarker:` does not fire and the marker is not selectable. 

This is caused by the current logic:

```objc
- (BOOL)mapView:(GMSMapView *)mapView didTapMarker:(GMSMarker *)marker {
  if ([_delegate respondsToSelector:@selector(clusterManager:didTapCluster:)] &&
      [marker.userData conformsToProtocol:@protocol(GMUCluster)]) {
    id<GMUCluster> cluster = marker.userData;
    [_delegate clusterManager:self didTapCluster:cluster];
    return YES;
  }

  if ([_delegate respondsToSelector:@selector(clusterManager:didTapClusterItem:)] &&
      [marker.userData conformsToProtocol:@protocol(GMUClusterItem)]) {
    id<GMUClusterItem> clusterItem = marker.userData;
    [_delegate clusterManager:self didTapClusterItem:clusterItem];
    return YES;
  }

  // Forward to _mapDelegate as a fallback.
  if ([_mapDelegate respondsToSelector:@selector(mapView:didTapMarker:)]) {
    return [_mapDelegate mapView:mapView didTapMarker:marker];
  }

  return NO;
}
```

So in my case, removing the return statements from the first two if-statements (where the cluster-delegates are checked) makes it working, because I let the `didTapMarker` decide whether or not the markers (not clusters (!)) should be selectable.

Thoughts very welcome!

**Comment by mountainvat** [comment]
It's an issue we're working on https://github.com/googlemaps/google-maps-ios-utils/issues/63.

**Comment by domesticmouse** [comment]
I believe this is now fixed with version 2.0.0 of the support library. See #63 for details. Thanks!
